### ISSUE TITLE
Parsing errors with a large number of fractional digits

### ISSUE THREAD

**Issue by mpfaff** [paupino/rust-decimal#461]
`Decimal.parse` gives incorrect results with some numbers with a large number of fractional digits.

Here are some examples:
```rust
println!("{}", Decimal::from_str(".00000000000000000000000000001").unwrap());
//1

println!("{}", Decimal::from_str(".10000000000000000000000000000").unwrap());
//10000000000000000000000000000
```

I think this has something to do with the absence of a 0 before the decimal point because when one is present this issue does not occur (as far as my testing has found).

**Comment by paupino** [comment]
`Decimal` doesn't have a `parse` function, however I believe you're using `from_str` so I'll answer from that perspective.

Good news: this should be fixed in `master` as part of #456! I'll likely release version `1.20` sometime this week which will include this fix.

e.g. On `master` this passes:

```rust
assert_eq!(Decimal::from_str(".00000000000000000000000000001").unwrap().to_string(), "0.0000000000000000000000000000");
assert_eq!(Decimal::from_str(".10000000000000000000000000000").unwrap().to_string(), "0.1000000000000000000000000000");
```

Version `1.19` currently fails on these tests returning `1` for the first result.

**Comment by mpfaff** [comment]
> `Decimal` doesn't have a `parse` function, however I believe you're using `from_str` so I'll answer from that perspective.

Yes, my mistake.

**Comment by mpfaff** [comment]
Perhaps I should open a new issue for this but is it intended that `0.1000000000000000000000000000` is printed instead of just `0.1`?

**Comment by paupino** [comment]
The generation of all of the extra `0`s is in fact purposeful - in this particular case it shows that it parsed the number with 29 decimals of precision. This is consistent with the .NET library which admittedly was a heavy inspiration early on: https://dotnetfiddle.net/dpicFO

**Comment by paupino** [comment]
This should now be formally resolved with the release of `1.20` - tests to confirm that are here: https://github.com/paupino/rust-decimal/blob/6d4c894a975e6694f47b7f4646547e5399b54d4b/tests/decimal_tests.rs#L2896-L2897.
